lucene/core/src/java/org/apache/lucene/index/DocumentsWriterPerThread.java

```diff
@@ -219,6 +222,9 @@ final void testPoint(String message) {
   /** Anything that will add N docs to the index should reserve first to
    *  make sure it's allowed. */
   private void reserveOneDoc() {
+    if (numDocsInRAM != 0 && hardMaxBytesPerDWPT < bytesUsed()) { // TODO should we check this after each field as well to also preempt within one doc?
```

Comment:
Or even per-token while inverting?  I think we already detect this today and throw aborting exception?  But we could fix that to instead throw a non-aborting exception, so the one document is deleted, but IW/DWPT are not aborted.  But we should do that later ...

Reply:
Could you reverse the conditional to `bytesUsed() > hardMaxBytesPerDWPT`?  I think it's more quickly readable in that order?